rest/blip_client_test.go

```diff
@@ -1105,25 +1060,29 @@ func (btc *BlipTesterCollectionClient) addCollectionProperty(msg *blip.Message)
 	return msg
 }
 
-func (btc *BlipTesterClient) getCollectionNameFromMessage(msg *blip.Message) (string, error) {
-	collectionIdx := msg.Properties[db.BlipCollection]
-	if len(btc.Collections) == 0 {
-		return "", nil
+func (bt *BlipTesterClient) addCollectionProperty(msg *blip.Message) *blip.Message {
+	if bt.nonCollectionAwareClient == nil {
+		msg.Properties[db.BlipCollection] = "0"
 	}
 
-	if collectionIdx == "" {
-		return "", fmt.Errorf("no collection given in %q message", msg.Profile())
-	}
+	return msg
+}
 
-	idx, err := strconv.Atoi(collectionIdx)
-	if err != nil {
-		return "", err
+func (btc *BlipTesterClient) getCollectionClientFromMessage(msg *blip.Message) *BlipTesterCollectionClient {
+	collectionIdx, exists := msg.Properties[db.BlipCollection]
+	if !exists {
+		require.NotNil(btc.rt.TB, btc.nonCollectionAwareClient)
```

Comment:
I'm on the fence about using require this deep into a testing utility, but if it significantly simplifies usage then I could be convinced.

Reply:
Many of it's callers panic when they call errors here, so I'd rather have a require.